### DISCUSSION TITLE
A new release?

### DISCUSSION THREAD

**Discussion by zhihaoy** [taocpp/PEGTL#148] [General]
I hope we can release more regularly so that we can update [vcpkg](https://github.com/Microsoft/vcpkg) accordingly.

**Comment by ColinH** [comment]
Just to be clear, do you mean that we should release more frequently and/or on a predetermined fixed schedule?

**Comment by zhihaoy** [comment]
More frequent than once every 3 months would be good.  Source-based ports systems (including FreeBSD ports) can easily catch up indeterminate very frequent releases, but if too long time without seeing a new release things (build scripts, patches, etc. in the port) can get decay.

**Comment by d-frey** [comment]
We usually release quite often as long as there is something new to release. The exception is major versions, as we do not want to break the API every couple of month.

That is the reason why the next release (3.0.0, switching to C++17) is in limbo for some time now, we need to come up with a plan which changes should still go in before a new major version. Colin and me will talk about it at the weekend, hopefully we can release 3.0.0 soon.

**Comment by zhihaoy** [comment]
I didn't released that 2.7 isn't C++17-only.  Okay, I filed an issue to ask around https://github.com/Microsoft/vcpkg/issues/5337.

**Comment by d-frey** [comment]
We've backported all features from the C++17 HEAD version back to the 2.x branch (unless they rely on C++17). Check the [Changelog](https://github.com/taocpp/PEGTL/blob/master/doc/Changelog.md) for the details. A new release of the 2.x branch will be done in the near future, we don't expect any major new features, just final details and documentation.

**Comment by d-frey** [comment]
We released version 2.8.0, which is still compatible with C++11. There is currently no estimate for a 3.0.0, the 2.x branch will be maintained as necessary.

**Comment by zhihaoy** [comment]
Thanks for the update.  I'm not using using 2.x but I can update vcpkg accordingly...

**Comment by d-frey** [comment]
OK, thanks!
